qemu/tests/watchdog.py

```diff
@@ -282,6 +283,51 @@ def hotplug_unplug_watchdog_device():
                 test.fail("Oops, find watchdog pci, unplug failed")
             logging.info("The WDT remove successfully")
 
+    def heartbeat_test():
+        """
+        Heartbeat test for i6300esb
+        Test steps:
+        1.Start VM with "-watchdog-action pause" CLI option
+        2.Set heartbeat value and reload the i6300esb module
+        3.Trigger wathchdog action through open /dev/watchdog
+        4.Ensure watchdog_action takes effect after $heartbeat.
+        """
+
+        del_module_cmd = params["del_module_cmd"]
+        reload_module_cmd = params["reload_module_cmd"]
+        _watchdog_device_check(test, session, watchdog_device_type)
+
+        error_context.context("set heartbeat value and reload the i6300esb "
+                              "module", logging.info)
+        session.cmd(del_module_cmd)
+        heartbeat = random.randint(1, 180)
+        session.cmd_output(reload_module_cmd % heartbeat)
+        _watchdog_device_check(test, session, watchdog_device_type)
+
+        _trigger_watchdog(session, trigger_cmd)
+
+        error_context.context("Judge watchdog fire time", logging.info)
+        start_time = time.time()
+        end_time = start_time + float(heartbeat)
+        while not vm.monitor.verify_status("paused"):
+            if time.time() > end_time + 2:
+                break
+            else:
+                time.sleep(1)
```

Comment:
Or remove line 315
```
while not vm.monitor.verify_status("paused"):
    if time.time() > end_time + 2:
        break
    time.sleep(1)
```

Reply:
i think line 315 is needed.
for example:
let's suppose now we're going into 'if' branch, and current time is > end_time + 2, now 'break'.
if line 315 is removed, you still need to sleep(1), which is pointless.

Reply:
@xiywang I believe after `break` if will go out to line 317, `time.sleep(1)` wont be executed.

Reply:
@vivianQizhu Sorry, my fault. It would goes to 317. Anyway, your suggestion and the original one have no big difference. Both way are OK to me.